src/firefly/js/ui/panel/CollapsiblePanel.jsx

```diff
@@ -113,28 +81,32 @@ export const FieldGroupCollapsible= memo( (props) => {
             isOpen: Boolean(viewProps.value === 'open'),
             onToggle: (isOpen) => fireValueChange({ value: isOpen ? 'open' : 'closed' })
         };
-    return ( <CollapsiblePanel {...newProps} /> );
+    return ( <CollapsiblePanelView {...newProps} /> );
 });
 
-CollapsiblePanel.propTypes = {
-    componentKey: PropTypes.string, // if need to preserve state and is not part of the field group
+CollapsiblePanelView.propTypes = {
     header: PropTypes.node,
     isOpen: PropTypes.bool,
     headerRoundCorner: PropTypes.number,
     headerStyle: PropTypes.object,
     contentStyle: PropTypes.object,
-    wrapperStyle: PropTypes.object,
+    style: PropTypes.object,
     borderStyle: PropTypes.number,
     onToggle: PropTypes.func,
     initialState: PropTypes.shape({
         value: PropTypes.string,  // 'open' or 'closed'
     })
 };
 
-CollapsiblePanel.defaultProps= {
-    headerRounderCorner: 0,
-    borderStyle: CollapseBorder.Oneborder,
-    isOpen: false
+CollapsiblePanel.propTypes = {
+    componentKey: PropTypes.string,
+    ...CollapsiblePanelView.propTypes
```

Comment:
Good. `componentKey` probably should be `isRequired`

Which don't you add `propTypes`  for `FieldGroupCollapsible` as well but in the case it would be
in that case instead of it should be `fieldGroupKey` as `isRequired`

Reply:
I don't think we can make `componentKey` because if we do so then for using the collapsible panel without component state, one has to use `CollapsiblePanelView` that doesn't have any state so user has to make sure of changing the `isOpen` prop being passed to the component each time `onToggle` is called. This made me realize that we should make `CollapsiblePanelView` as internal component by removing the `export`. What do you think?

I have added `propTypes` for `FieldGroupCollapsible` too.

Reply:
It should be required. We don't have a use case for a simple state. We had that for awhile but it is lost every time you unrender it. There for the two was to use it is with `componentKey` or as part of a field group. Also- it you let it be optional you will put and `undefined` key in the store.

Reply:
Ohk